domain-models-runtime/src/test/resources/mockdata/poLines.json

```diff
@@ -1,6 +1,6 @@
+{"id": "90fb53f2-cdf1-11e8-a8d5-f2801f1b9fd1","edition": "First edition","checkin_items": false,"instance_id": "2c287d1c-0e30-4d61-566b-9c9adc573a99","agreement_id": "2c287d1c-0r60-6g81-258b-7b5aaf573a58","purchase_order_id": "05bdf3c8-01f0-4ddb-bd6c-6efd465f9e33","acquisition_method": "Purchase","adjustment": "496336cf-eb47-4e09-9e28-e35aa464adb7","alerts": [],"cancellation_restriction": false,"cancellation_restriction_note": "","claims": ["47c5d666-cabe-11e8-a8d5-f2801f1b9fd1"],"collection": false,"cost": "47c5d792-cabe-11e8-a8d5-f2801f1b9fd1","description": "","details": "47c5d792-01be-11e9-a8d5-f2801f1b9fd1","donor": "","eresource": "47c5d801-cabe-11e8-a8d5-f2801f1b9fd1","fund_distribution": ["4de2c888-cde3-11e8-a8d5-f2801f1b9fd1"],"location": "47c5d9ea-cabe-11e8-a8d5-f2801f1b9f02","order_format": "P/E Mix","owner": "Cushing","payment_status": "Pending","physical": "90fb71e8-cdf1-11e8-a8d5-f2801f1b9fd1","po_line_description": "","po_line_number": "1","po_line_workflow_status": "Open","publication_date": "2016","publisher": "ro*co films","receipt_date": null,"receipt_status": "Awaiting Receipt","reporting_codes": [],"requester": "jcobb","rush": false,"selector": "mreynolds","source": "4b4b2f41-c5c6-4280-858e-33fba452a134","tags": ["dvd", "streaming"],"title": "Acquisitions Rules the Library","vendor_detail": "90fb8a2a-cdf1-11e8-a8d5-f2801f1b9fd1", "metadata":{ "createdDate": "2018-07-19T00:00:00Z",  "createdByUserId": "28d100f4-d137-11e8-a8d5-f2801f1b9fd1"}}
+{"id": "a0fb5514-cdf1-11e8-a8d5-f2801f1b9fd1","edition": "Second edition","checkin_items": false,"instance_id": "2e6b4b90-0e30-4d61-566b-9c9adc573a99","agreement_id": "2e6b4b90-0r60-6g81-258b-7b5aaf573a58","purchase_order_id": "07f65192-44a4-483d-97aa-b137bbd96390","acquisition_method": "Purchase","adjustment": "596336cf-eb47-4e09-9e28-e20aa464adb7","alerts": [],"cancellation_restriction": false,"cancellation_restriction_note": "","claims": ["57c5d666-cabe-11e8-a8d5-f2801f1b9fd1"],"collection": true,"cost": "57c3d792-cabe-11e8-a8d5-f2801f1b9fd1","description": "","details": "57c5d792-cabe-11e9-a8d5-f2801f1b9fd1","donor": "","eresource": "57c5d803-cabe-11e8-a8d5-f2801f1b9fd1","fund_distribution": ["5de2cad6-cde3-11e8-a8d5-f2801f1b9fd1"],"location": "57c5d9ea-cabe-11e8-a8d5-f2801f1b9f04","order_format": "Electronic Resource","owner": "Technical Services Department","payment_status": "Awaiting Payment","physical": "a0fb7332-cdf1-11e8-a8d5-f2801f1b9fd1","po_line_description": "","po_line_number": "80","po_line_workflow_status": "Closed","publication_date": "2019","publisher": "American Chemical Society","receipt_date": null,"reporting_codes": [],"requester": "","rush": false,"selector": "Buchanan","source": "5b4b6f41-c5c6-4280-858e-33fba452a334","tags": [],"title": "American Chemical Society Web Editions","vendor_detail": "a0fb8b56-cdf1-11e8-a8d5-f2801f1b9fd1", "metadata":{ "createdDate": "2018-07-18T00:00:00Z",  "createdByUserId": "28d102a2-d137-11e8-a8d5-f2801f1b9fd1"}}
```

Comment:
Can you add a `count` field of type integer again (and add it to Poline.java) to make sure this is parsed as a field of a regular record and not as the count record?
You may remove other fields if you like.

Reply:
No. I'd rather not do that. Then my case is not illustrated.. count field will be seen as normal record *and* the getInteger(COUNT_FIELD) is not executed (because mapping works for the Poline Object).

Reply:
Ok, can you add another unit test for this use case:
The RAML JSON file declares a record with "id" (uuid string) and "count" (integer) field.
This results in a POJO with id and count field.
Either RMB cannot handle this because of the name clash. Then it should be documented and RMB should reject such a RAML JSON entity with an error message.
Or RMB can properly handle this because RMB can distinguish a POJO count field from the query count field. Then we need a unit test for this case.

Reply:
The bottomline is.. The count object can NOT be distinguished from another count object. The solution is obvious: change SQL to just not return that object at all. It's pointless, now that qe have a separate counting process anyway. Doing in next PR which simplifies a lot.

Reply:
Given a JSON object that can be deserilized to TWO pojos, we're in trouble. Let's not go there.